src/NotificationCenter/Notifications.hs

```diff
@@ -441,18 +445,57 @@ removeNotiFromDistList tState id = do
     return False
   return ()
 
+-- | Adjusts the position of all displayed notifications so they follow standardized placement rules.
+readjustNotificationPositions :: Config -> TVar NotifyState -> IO ()
+readjustNotificationPositions config tState = do
+  sortedDisplayedPopups <- sortOn _dNotiTop . filter (not . _dHasCustomPosition) . notiDisplayingList <$> readTVarIO tState
```

Comment:
The readTVarIO state reads a shared memory location. I think for performance reasons (and readability) it makes sense to get the state once, before

```suggestion
  state <- readTVarIO tState
  sortedDisplayedPopups <- sortOn _dNotiTop . filter (not . _dHasCustomPosition) . notiDisplayingList state
```